File: programmers/DFS_1.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        int start = numbers[0]; //target이랑 같아지는지 확인할 변수, 계속 증감 함
        
        answer = dfs(1,start,numbers,target)+dfs(1,-start,numbers,target);
        return answer;
    }
    int dfs(int index, int start, int[] numbers, int target){
        if(index == numbers.length){//돌만큼 다 돌았다
            if(start==target){
                return 1; //카운트에 1을 더함
            }
            else{
                return 0; //카운트에 아무것도 안더함(조건과 맞지 않음)
            }
        }
        int result = 0;
        result += dfs(index+1, start+numbers[index],numbers, target);
        result += dfs(index+1, start-numbers[index],numbers, target);
        return result;
    }
}

File: programmers/DFS_1.py
#타겟넘버
def solution(numbers, target):
    answer = 0
    start = numbers[0] #타겟이랑 같아지는지 확인할 변수(현재값)
    answer = dfs(1,start,numbers, target)+dfs(1,-start,numbers, target)
    
    return answer
def dfs(index, start,numbers, target):
        if(index == len(numbers)):#돌만큼 다 돌았다
            if(start == target):
                return 1 #카운트+1(타겟이랑 맞춤)
            else:
                return 0 #카운트 x(조건에 맞지 않음)
        result = 0
        #다음 값을 더하고 빼는 각 경우의 수로 가지침
        result += dfs(index+1, start+numbers[index],numbers, target)
        result += dfs(index+1, start-numbers[index],numbers, target)
        return result
